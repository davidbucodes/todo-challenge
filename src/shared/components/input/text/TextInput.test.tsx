import { composeStories } from "@storybook/react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import * as stories from "./TextInput.stories";

const { Default, WithLabel, WithPlaceholder, WithInitialValue } =
  composeStories(stories);

const getTextInput = () => screen.getByRole("textbox");

describe("TextInput", () => {
  describe("Default", () => {
    it("should render correctly", () => {
      const { container } = render(<Default />);
      const textInput = getTextInput();
      expect(container).toMatchSnapshot();
      expect(textInput).not.toBeNull();
    });

    describe("onInputValueChanged", () => {
      it("should emit onInputValueChanged event", async () => {
        const onValueChange = jest.fn();
        render(<Default onValueChange={onValueChange} />);
        const textInput = getTextInput();
        const textToType = "text";

        await userEvent.type(textInput, textToType);

        expect(onValueChange).toBeCalledTimes(textToType.length);
        expect(onValueChange).toBeCalledWith(textToType);
      });
    });
  });

  describe("WithLabel", () => {
    it("should render correctly", () => {
      const { container } = render(<WithLabel />);
      const textInput = getTextInput();
      const label = WithLabel.args.label as string;
      const textInputLabel = screen.getByText(label);
      expect(container).toMatchSnapshot();
      expect(textInput).not.toBeNull();
      expect(textInputLabel).not.toBeNull();
    });
  });

  describe("WithPlaceholder", () => {
    it("should render correctly", () => {
      const { container } = render(<WithPlaceholder />);
      expect(container).toMatchSnapshot();
      const textInput = getTextInput();
      expect(textInput).not.toBeNull();
    });
  });

  describe("WithInitialValue", () => {
    it("should render correctly", () => {
      const { container } = render(<WithInitialValue />);
      expect(container).toMatchSnapshot();
      const textInput = getTextInput();
      expect(textInput).not.toBeNull();
    });
  });
});
