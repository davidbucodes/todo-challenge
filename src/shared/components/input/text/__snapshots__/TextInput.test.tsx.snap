// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput Default should render correctly 1`] = `
<div>
  <div
    class="sc-bgqQcB"
  >
    <input
      class="sc-gTRrQi"
      id="id"
      type="text"
      value=""
    />
  </div>
</div>
`;

exports[`TextInput WithInitialValue should render correctly 1`] = `
<div>
  <div
    class="sc-bgqQcB"
  >
    <input
      class="sc-gTRrQi"
      id="id"
      type="text"
      value="initialValue"
    />
  </div>
</div>
`;

exports[`TextInput WithLabel should render correctly 1`] = `
<div>
  <div
    class="sc-bgqQcB"
  >
    <label
      class="sc-ewnqHT exFVTf"
      for="id"
    >
      Label
    </label>
    <input
      class="sc-gTRrQi"
      id="id"
      type="text"
      value=""
    />
  </div>
</div>
`;

exports[`TextInput WithPlaceholder should render correctly 1`] = `
<div>
  <div
    class="sc-bgqQcB"
  >
    <input
      class="sc-gTRrQi"
      id="id"
      placeholder="placeholder"
      type="text"
      value=""
    />
  </div>
</div>
`;
